### PR TITLE
HV-763 Use @FormatWith consistently in logging API

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/HV-763

### CODE REVIEW

**Comment by gsmet** [comment]
Jenkins, please retest this

**Comment by gsmet** [comment]
Jenkins, retest this please

File: engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java [hunk 1]
```diff
@@ -228,10 +246,10 @@ UnexpectedTypeException getNoValidatorFoundForTypeException(String constraintTyp
 	@Message(id = 63, value = "Unable to instantiate %s.")
 	ValidationException getUnableToInstantiateException(String className, @Cause Exception e);
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java]: What about this one?

File: engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java [hunk 2]
```diff
@@ -228,10 +246,10 @@ UnexpectedTypeException getNoValidatorFoundForTypeException(String constraintTyp
...
 
-	ValidationException getUnableToInstantiateException(Class<?> clazz, @Cause Exception e);
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java]: Hum, no `@Message` annotation? Something is odd here.

File: engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java [hunk 3]
```diff
@@ -387,11 +411,13 @@ UnexpectedTypeException getNoValidatorFoundForTypeException(String constraintTyp
...
 	@Message(id = 118, value = "Unable to cast %s (with element kind %s) to %s")
-	ClassCastException getUnableToNarrowNodeTypeException(String actualDescriptorType, ElementKind kind, String expectedDescriptorType);
+	// TODO
+	ClassCastException getUnableToNarrowNodeTypeException(@FormatWith(ClassObjectFormatter.class) Class<?> actualDescriptorType, ElementKind kind,
+			@FormatWith(ClassObjectFormatter.class) Class<?> expectedDescriptorType);
 
 	@LogMessage(level = INFO)
 	@Message(id = 119, value = "Using %s as parameter name provider.")
-	void usingParameterNameProvider(String parameterNameProviderClassName);
+	void usingParameterNameProvider(@FormatWith(ClassObjectFormatter.class) Class<ParameterNameProvider> parameterNameProviderClass);
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java]: Wondering, wouldn't it technically be `Class<? extends ParameterNameProvider>`?

File: engine/src/main/java/org/hibernate/validator/internal/util/logging/formatter/CollectionOfClassesObjectFormatter.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Used with JBoss Logging to display collection of class names in log messages.
+ *
+ * @author Guillaume Smet
+ */
+public class CollectionOfClassesObjectFormatter {
+
+	private final String stringRepresentation;
+
+	public CollectionOfClassesObjectFormatter(Collection<Class<?>> classes) {
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/util/logging/formatter/CollectionOfClassesObjectFormatter.java]: Or  this:

    stringRepresentation = classes.stream()
        .map( c -> c.getName() )
        .collect( Collectors.joining( ", " ) );

Sorry, couldn't resist :)

**Comment by gunnarmorling** [comment]
@gsmet Looks good, added some minor notes. I reckon you don't want to backport it given it's using Java 7 syntax already?

File: engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java (hunk 3 above)
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java]: Sure. There are several like that. But the caller is casting the argument to Class<ParameterNameProvider>, that's why I did it like that. Maybe we should fix the caller?

File: engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java (hunk 1 above)
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java]: IIRC, the error is called when it's unable to instantiate the class. Thus we can't have a class parameter :).

File: engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java (hunk 2 above)
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java]: Yup, I was wondering if there was some magic to get the message from the previous one (as I would expect them to be the same)?

**Comment by gsmet** [comment]
Yes, I don't think we should backport it. It does not bring anything to the user.

File: engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java (hunk 2 above)
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java]: I confirm there's some magic involved: the second method reuses the message of the first method. So it's weird but OK.

File: engine/src/main/java/org/hibernate/validator/internal/util/logging/formatter/CollectionOfClassesObjectFormatter.java (hunk 4 above)
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/util/logging/formatter/CollectionOfClassesObjectFormatter.java]: Gunnar happy camper, it is!

File: engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java (hunk 3 above)
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java]: I fixed the casts in the callers and the Log methods.

**Comment by gunnarmorling** [comment]
Rebased and applied. Added one more commit to get rid of that duplicated method (it was about instantiating not loading a class by name, so we had the class object already).

**Pull request closed without merging**